mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -570,6 +570,146 @@ static LogicalResult generateLoopNest(
   return rewriter.notifyMatchFailure(loc, "unhandled loop type");
 }
 
+static LogicalResult
```

Comment:
Could we replace this wil `FailureOr<SmallVector<Value>>` . Trying to avoid return by reference.

Reply:
done